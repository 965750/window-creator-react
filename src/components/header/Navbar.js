import React, { Component } from 'react'
import styles from './Navbar.module.scss'
import LanguageSwitcher from './LanguageSwitcher'

export default class Navbar extends Component {
    render() {
        return (
            <div className={`${styles.navbar} bg-white shadow`}>
                NAVBAR
                <LanguageSwitcher />
            </div>
        )
    }
}
