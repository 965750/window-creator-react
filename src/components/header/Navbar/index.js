import PropTypes from 'prop-types'
import React from 'react'
import styles from './Navbar.module.scss'

import LanguageSwitcher from '../LanguageSwitcher'
import LoggedInLinks from '../LoggedInLinks/'
import LoggedOutLinks from '../LoggedOutLinks/'

const Navbar = ({ auth }) => (
  <div
    className={`${styles.navbar} px-3 sm:px-5 flex bg-white shadow space justify-between`}
  >
    <LanguageSwitcher />
    {!auth.isEmpty ? <LoggedInLinks /> : <LoggedOutLinks />}
  </div>
)

Navbar.propTypes = {
  auth: PropTypes.object,
}

export default Navbar
