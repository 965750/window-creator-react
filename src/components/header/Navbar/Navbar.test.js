import '@testing-library/jest-dom'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from "@testing-library/react"

import Navbar from './index'
import messages from "../../../messages-i18n"
import thunk from 'redux-thunk';
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl'
  },
  firebase: {
    profile: {
      windows: [1, 2, 3],
    }
  }
}
const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => {
  return {
    ...render(<Provider store={store}>{component}</Provider>)
  }
}

describe('Navbar', () => {
  let wrapper
  const initialProps = {
    auth: {
      isEmpty: false
    }
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale='en'
        messages={messages[store.getState().theme.lang]}
      >
        <Navbar {...props}/>
      </IntlProvider>
    )
  }

  afterEach(cleanup)

  it('should match logged in snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should match logged out snapshot', () => {
    setUp({
      auth: {
        isEmpty: true
      }
    })

    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
