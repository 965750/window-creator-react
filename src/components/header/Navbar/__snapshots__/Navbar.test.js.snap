// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match logged in snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar px-3 sm:px-5 flex bg-white shadow space justify-between"
  >
    <div
      class="flex"
    >
      <p
        class="select__label mr-5"
      >
        Wybierz język:
      </p>
      <div
        class="select__wrapper relative relative mt-lg"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down fa-w-14 select__arrow absolute right-10"
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
          />
        </svg>
        <select
          class="select h-full w-full pl-3 border border-gullGray rounded bg-white outline-none"
          data-testid="select"
        >
          <option
            value="pl"
          >
            Polski
          </option>
          <option
            value="en"
          >
            Angielski
          </option>
        </select>
      </div>
    </div>
    <div
      class="flex relative"
    >
      <button
        class="z-40 uppercase mr-5 cursor-pointer"
      >
        <span
          class="hover:text-malachite"
        >
          Wczytaj
        </span>
      </button>
      <button
        class="uppercase mr-3 md:mr-5 hover:text-malachite cursor-pointer"
      >
        Wyloguj
      </button>
      <div
        class="avatar z-30 mt-md rounded-full bg-mountainMeadow cursor-pointer"
        data-testid="avatar"
      >
        <p
          class="uppercase text-center pt-1 leading-loose text-white text-xl font-bold"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navbar should match logged out snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar px-3 sm:px-5 flex bg-white shadow space justify-between"
  >
    <div
      class="flex"
    >
      <p
        class="select__label mr-5"
      >
        Wybierz język:
      </p>
      <div
        class="select__wrapper relative relative mt-lg"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down fa-w-14 select__arrow absolute right-10"
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
          />
        </svg>
        <select
          class="select h-full w-full pl-3 border border-gullGray rounded bg-white outline-none"
          data-testid="select"
        >
          <option
            value="pl"
          >
            Polski
          </option>
          <option
            value="en"
          >
            Angielski
          </option>
        </select>
      </div>
    </div>
    <div
      class="uppercase flex"
    >
      <a
        aria-current="page"
        class="text-osloGray hover:text-malachite flex self-center cursor-pointer sm:mr-10 mr-4 link--active"
        href="/"
      >
        Logowanie
      </a>
      <a
        class="text-osloGray hover:text-malachite flex self-center cursor-pointer"
        href="/register"
      >
        Rejestracja
      </a>
    </div>
  </div>
</DocumentFragment>
`;
